view/frontend/web/js/utils/when-defined.js

```diff
@@ -29,45 +29,47 @@ define([], function () {
      * @param {Object} object to check for property definition
      * @param {number|string} property that is expected to be defined on {@see object}
      * @param {Function} callback function to be called when {@see property} gets defined on {@see object}
-     * @param {null} key deprecated parameter used for setting multiple callbacks per property
+     * @param {string} key used to prevent setting duplicate callbacks 
      */
-    function whenDefined(object, property, callback, key = null) {
```

Comment:
We need a key for cases when we want to add two callbacks to the same property, right?
I don't know JS well, but wonder if it's possible to check if we call the function with the same callback or with a different one?

Reply:
I don't have a definite answer here. There are probably ways of doing it. Maybe if we always first define the function and pass the callback as a named functions instead of declaring an anonymous one inline, as we usually do. Seems out of scope of this particular problem and also not too practical.